src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -281,20 +281,38 @@
     (future-cancel command-loader))
   context)
 
-(defn initialize-schema
-  "Ensure the database is migrated to the latest version, and warn if
-  it's deprecated, log and exit if it's unsupported.
-
-  This function will return true iff any migrations were run."
-  [db-conn-pool config]
+(defn db-unsupported-msg
+  "Returns a message describing which databases are supported."
+  []
+  (str (trs "PostgreSQL DB versions older than {0}.{1} are no longer supported."
+            (first scf-store/oldest-supported-db)
+            (second scf-store/oldest-supported-db))
+       " "
+       (trs "Please upgrade Postgres and restart PuppetDB.")))
+
+(defn shutdown-if-db-not-supported [shutdown-on-error]
+  (let [oldest scf-store/oldest-supported-db]
+    (when (sutils/db-version-older-than? oldest)
+      (let [msg (db-unsupported-msg)
+            attn (utils/attention-msg msg)]
+        (utils/println-err attn)
+        (log/error attn)
+        (shutdown-on-error #(throw (Exception. msg)))))))
```

Comment:
Yeah, this is really icky. Not much to be done about it now; maybe a comment above it would be good, informing whomever reads this next of just what we were thinking.